### PR TITLE
Fixed settings in guest window

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/24242

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

same from issue

### CODE REVIEW

File: browser/resources/settings/brave_routes.js [hunk 1]
```diff
@@ -40,8 +40,9 @@ export default function addBraveRoutes(r) {
     r.BRAVE_WALLET = r.BASIC.createSection('/wallet', 'wallet')
     r.BRAVE_WALLET_NETWORKS = r.BRAVE_WALLET.createChild('/wallet/networks');
   }
-
-  r.BRAVE_HELP_TIPS = r.ADVANCED.createSection('/braveHelpTips', 'braveHelpTips')
+  if (!isGuest) {
+    r.BRAVE_HELP_TIPS = r.ADVANCED.createSection('/braveHelpTips', 'braveHelpTips')
+  }
```
Comment (simonhong) [inline browser/resources/settings/brave_routes.js:45]: nit: How about below for error logging?
```
  if (r.ADVANCED) {
    r.BRAVE_HELP_TIPS = r.ADVANCED.createSection('/braveHelpTips', 'braveHelpTips')
  } else if (!isGuest) {
    console.error('[Brave Settings Overrides] Routes: could not find ADVANCED page')
  }
```

**Review by simonhong — APPROVED** [review]
++ with one nit. Just a suggestion. It's up to you.

File: browser/resources/settings/brave_routes.js (hunk 1 above)
Comment (sangwoo108) [inline browser/resources/settings/brave_routes.js:45]: nit: It seems to work fine as is but I think we can do this similar with other cases.

```suggestion
  if (r.ADVANCED) {
    r.BRAVE_HELP_TIPS = r.ADVANCED.createSection('/braveHelpTips', 'braveHelpTips')
  } else if (!isGuest) {
     console.error('[Brave Settings Overrides] Could not find ACVANCED page', r)
  }
```

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: browser/resources/settings/brave_routes.js (hunk 1 above)
Comment (sangwoo108) [reply browser/resources/settings/brave_routes.js:45]: 5 min late 😹

Comment (simonhong) [reply browser/resources/settings/brave_routes.js:45]: funny!  wrote same comment :)

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.44.4 Chromium: 104.0.5112.69 (Official Build) nightly (64-bit)
-- | --
Revision | 7ce2902023c722af8564068e6b26e934b83fd774-refs/branch-heads/5112@{#1213}
OS | Windows 11 Version 21H2 (Build 22000.832)
```

* ensured that `brave://settings` opens without any issues via `Private` windows
* ensured that `brave://settings` opens without any issues via `Tor` windows
* ensured that `brave://settings` opens without any issues via `Guest` windows
* ensured that `brave://settings` opens without any issues via `Normal` windows

**`Example (Private)`** | **`Example (Tor)`** | **`Example (Guest)`** | **`Example (Normal)`**
-------------------------|----------------------|------------------------|----------------------------
private | tor | guest | normal

**Pull request merged**